classes/Employee.php

```diff
@@ -233,16 +233,26 @@ protected function updateTextDirection()
 
     protected function saveOptin()
     {
-        if ($this->optin && !defined('PS_INSTALLATION_IN_PROGRESS')) {
+        if (!defined('PS_INSTALLATION_IN_PROGRESS')) {
             $language = new Language($this->id_lang);
-            $params = http_build_query(array(
-                'email' => $this->email,
-                'method' => 'addMemberToNewsletter',
-                'language' => $language->iso_code,
-                'visitorType' => 1,
-                'source' => 'backoffice'
-            ));
-            Tools::file_get_contents('http://www.prestashop.com/ajax/controller.php?'.$params);
+            if ($this->optin == 1) {
+                $params = http_build_query(array(
+                    'email' => $this->email,
+                    'method' => 'addMemberToNewsletter',
+                    'language' => $language->iso_code,
+                    'visitorType' => 1,
+                    'source' => 'backoffice'
+                ));
+            } else {
+                $params = http_build_query(array(
+                    'email' => $this->email,
+                    'method' => 'removeMemberToNewsletter', // We don't know the method
```

Comment:
Only prestashop know the good param['method'] and he probably not exist

Reply:
I'm not aware of method to remove member form news letter, sounds like complicated to accept the contribution without this method.

Ping @Quetzacoalt91

Reply:
I don't know either, I have asked internally.

Reply:
Ok, problem it's not only merchants it's also my employees.

But i undunderstand it's hard to make it.

I will close this PR un put a new one only with https.

You are informed :)